# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../hte/dis.patch
# Copyright (C) 2006 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -u -r1.26.4.14 htdisasm.cc
--- ./htdisasm.cc	1 Dec 2006 21:26:54 -0000	1.26.4.14
+++ ./htdisasm.cc	21 Dec 2006 15:16:29 -0000
@@ -421,9 +421,9 @@
 bool ht_disasm_sub::getline(char *line, const LINE_ID line_id)
 {
 	if (line_id.id2) return false;
-	uint32 ofs = line_id.id1;
-	byte buf[15];
-	int c = MIN(16, (int)(fofs+fsize-ofs));
+	uint64 ofs = line_id.id1;
+	byte buf[16];
+	int c = MIN(16, sint64(fofs+fsize-ofs));
 	if (c <= 0) return false;
 	file->seek(ofs);
 	c = file->read(buf, c);
